dotCMS/src/main/java/com/dotcms/util/content/json/PopulateContentletAsJSONUtil.java

```diff
@@ -0,0 +1,398 @@
+    private final String OPEN_CURSOR_MSSQL = "OPEN missingContentletAsJSONCursor";
+    private final String CLOSE_CURSOR = "CLOSE missingContentletAsJSONCursor";
+    private final String DEALLOCATE_CURSOR_MSSQL = "DEALLOCATE missingContentletAsJSONCursor";
+
+    protected static final int MAX_UPDATE_BATCH_SIZE = Config.getIntProperty("task.230320.maxupdatebatchsize", 100);
+    protected static final int MAX_CURSOR_FETCH_SIZE = Config.getIntProperty("task.230320.maxcursorfetchsize", 100);
+    protected final List<Params> paramsUpdate = new ArrayList<>();
+
+    public PopulateContentletAsJSONUtil() {
+        this.contentletJsonAPI = APILocator.getContentletJsonAPI();
+    }
+
+    /**
+     * Finds all the contentlets that need to be updated with the contentlet_as_json column for a given
+     * assetSubtype (Content Type).
+     *
+     * @param assetSubtype Asset subtype (Content Type) to filter the contentlets to process, if null then all
+     *                     the contentlets will be processed.
+     * @throws SQLException
+     * @throws DotDataException
+     * @throws IOException
+     */
+    public void populateForAssetSubType(String assetSubtype) throws SQLException, DotDataException, IOException {
+        Logger.info(this, String.format("Populate Contentlet as JSON task started for asset subtype [%s]", assetSubtype));
+        populate(assetSubtype, null);
+    }
+
+    /**
+     * Finds all the contentlets that need to be updated with the contentlet_as_json column excluding the contentles
+     * of a given assetSubtype (Content Type).
+     *
+     * @param assetSubtype Asset subtype (Content Type) use to exclude contentlets of that given type from the query.
+     * @throws SQLException
+     * @throws DotDataException
+     * @throws IOException
+     */
+    public void populateExcludingAssetSubType(String assetSubtype) throws SQLException, DotDataException, IOException {
+        Logger.info(this, String.format("Populate Contentlet as JSON task started excluding asset subtype [%s]", assetSubtype));
+        populate(null, assetSubtype);
+    }
+
+    /**
+     * Finds all the contentlets that need to be updated with the contentlet_as_json column for the given
+     * assetSubtype and excludingAssetSubtype.
+     *
+     * @param assetSubtype          Optional asset subtype (Content Type) to filter the contentlets to process, if null then all
+     *                              the contentlets will be processed unless the excludingAssetSubtype is provided.
+     * @param excludingAssetSubtype Optional asset subtype (Content Type) use to exclude contentlets from the query
+     * @throws SQLException
+     * @throws DotDataException
+     * @throws IOException
+     */
+    private void populate(@Nullable String assetSubtype, @Nullable String excludingAssetSubtype)
+            throws SQLException, DotDataException, IOException {
+
+        final StopWatch stopWatch = new StopWatch();
```

Comment:
I think we have an annotation for the stopWatch

Reply:
Done